Loop/Managers/DeviceDataManager.swift

```diff
@@ -320,14 +361,19 @@ final class DeviceDataManager {
         return Manager.init(rawState: rawState) as? PumpManagerUI
     }
 
-    private func processCGMResult(_ manager: CGMManager, result: CGMResult) {
-        switch result {
+    private func processCGMReadingResult(_ manager: CGMManager, readingResult: CGMReadingResult) {
+        switch readingResult {
         case .newData(let values):
             log.default("CGMManager:%{public}@ did update with %d values", String(describing: type(of: manager)), values.count)
 
             loopManager.addGlucoseSamples(values) { result in
                 self.log.default("Asserting current pump data")
                 self.pumpManager?.ensureCurrentPumpData(completion: nil)
+                if !values.isEmpty {
```

Comment:
If we do not invoke `self.cgmStalenessMonitor.cgmGlucoseSamplesAvailable` when `values.isEmpty` won't this prevent `cgmDataIsStale` from updating correctly?

Reply:
After looking into the implementation of `cgmGlucoseSamplesAvailable` I see why this check can be done here and it won't cause problems. However, I think the internal details of `cgmGlucoseSamplesAvailable` (that an empty values won't change anything because any previous CGM staleness timer would still eventually fire) are being exposed here. Consider removing the `isEmpty` check here and adding it into `cgmGlucoseSamplesAvailable`.

Reply:
But this is now just a [nit]. :)

Reply:
I don't think any internal details of cgmGlucoseSamplesAvailable are being exposed. I'm simply not calling it because there are no samples available.  Staleness has to be a timer because we might not get any sort of ticks from the CGM.

Reply:
It seems odd to me to send a cgmGlucoseSamplesAvailable message to an object when no samples are available.